polyaxon/templates/secrets.yaml

```diff
@@ -51,7 +51,9 @@ data:
   {{- end }}
   {{- if .Values.privateRegistries }}
   {{- range $index, $val := .Values.privateRegistries }}
-  POLYAXON_PRIVATE_REGISTRY_{{ $index }}: {{ $val | b64enc | quote }}
+  POLYAXON_PRIVATE_REGISTRY_{{ $index }}_USER: {{ $val.username | b64enc | quote }}
```

Comment:
That was my initial thinking, but I did not spend too much time thinking about it, I also did not think that you would spend time trying the approach, kudos!

One thing is, I left from my previous comment is that the dict needs to be a spec to avoid changing code in other places:

```python
registry_dict = config.get_dict(key, is_secret=True)
registry_spec = UriSpec(**registry_dict)
```

Reply:
To be able to keep both we can do the following:

```python
try:
    registry_dict = config.get_dict(key, is_secret=True)
    registry_spec = UriSpec(**registry_dict)
except RheaError:
    registry_spec = config.get_string(key, is_secret=True)
    registry_spec = config.parse_uri_spec(registry_spec)
```

Reply:
I'll commit the last code to the platform. If you could check this and integrate to the next release 0.3.0, will be awesome !

Reply:
I will do, I am confident it will be part of next release.